server/app/App.ts

```diff
@@ -94,13 +96,15 @@ export class App {
 
   public async start(): Promise<void> {
     try {
-      if (!this.app) {
+      if (!this.server) {
         throw new Error('app failed to initialize');
       }
-      const port = this.config.get<number>('server.port');
 
-      this.listener = this.app.listen(port, () => {
-        this.logger.info(`Service booted on port ${port}`);
+      const configPort = this.config.get<number>('server.port');
+
+      this.port = typeof this.port !== 'undefined' ? this.port : configPort;
```

Comment:
Testing `port?: number` against a string `'undefined'`?

Reply:
typeof not string, if no port is passed its typeof matches 'undefined'

Reply:
ahh gotcha, my bad